common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,136 @@
+"""
+The Python API layer of the country access settings. Essentially the middle tier of the project, responsible for all
+business logic that is not directly tied to the data itself.
+
+This API is exposed via the middleware(emabargo/middileware.py) layer but may be used directly in-process.
+
+"""
+import logging
+import pygeoip
+
+from django.core.cache import cache
+from django.conf import settings
+from django.http import HttpResponseRedirect, HttpResponseForbidden
+from embargo.models import CountryAccessRule, IPFilter, RestrictedCourse
+
+log = logging.getLogger(__name__)
+
+def _check_ip_lists(ip_addr):
+    """
+    Check whether the user is embargoed based on the IP address.
+
+    Args:
+        ip_addr (str): The IP address the request originated from.
+
+    Returns:
+        A True if the user is not blacklisted, otherwise False
+
+    """
+    # If blacklisted, immediately fail
+    if ip_addr in IPFilter.current().blacklist_ips:
+        return False
+
+    # If we're white-listed, then allow access
+    if ip_addr in IPFilter.current().whitelist_ips:
+        return True
+
+    return True
+    # If none of the other checks caught anything,
+    # implicitly return True to indicate that the user can access the course
+
+
+def _check_user_country(country_code, course_id=u""):
```

Comment:
Maybe remove this function and replace it with direct calls to `CountryAccessRule.check_country_access()`?